src/Bicep.Core/Extensions/IPositionableExtensions.cs

```diff
@@ -8,9 +8,18 @@ namespace Bicep.Core.Extensions
     public static class IPositionableExtensions
     {
         public static TextSpan ToZeroLengthSpan(this IPositionable positionable)
-            => new TextSpan(positionable.Span.Position, 0);
+            => new(positionable.Span.Position, 0);
+
+        public static int GetPosition(this IPositionable positionable)
+            => positionable.Span.Position;
 
         public static int GetEndPosition(this IPositionable positionable)
             => positionable.Span.Position + positionable.Span.Length;
+
+        public static bool IsOverlapping(this IPositionable positionable, int position)
+            => positionable.GetPosition() <= position && position <= positionable.GetEndPosition();
```

Comment:
>positionable.GetPosition() <= position && position <= positionable.GetEndPosition(); [](http://example.com/codeflow?start=15&length=84)

We have `ContainsInclusive` on `TextSpan`. Should we call it here and potentially rename to match?